test/jasmine/tests/template_test.js

```diff
@@ -141,6 +141,33 @@ describe('makeTemplate', function() {
 
         expect(template).toEqual(expected);
     });
+
+    it('works on DOM element', function(done) {
+        var mock = Lib.extendDeep({}, scatterFillMock);
+        var gd = createGraphDiv();
+
+        Plotly.newPlot(gd, mock)
+      .then(function() {
+          var template = Plotly.makeTemplate(gd);
+          delete(template.layout.xaxis);
+          delete(template.layout.yaxis);
```

Comment:
For this test to pass, some elements needed to be deleted (namely `layout.xaxis` and `layout.yaxis`. The result is not exactly the same as calling `makeTemplate` with the figure object. Isn't this problematic? 🤔

Reply:
Yep, that's to be expected 👍